src/blockstorage/blockstorage.cpp

```diff
@@ -689,7 +689,8 @@ bool FlushStateToDiskInternal(CValidationState &state,
             size_t nTrimSize = nCoinCacheMaxSize * .90;
             if (nCoinCacheMaxSize - nMaxCacheIncreaseSinceLastFlush > nTrimSize)
             {
-                nTrimSize = nCoinCacheMaxSize - nMaxCacheIncreaseSinceLastFlush;
+                if (nCoinCacheMaxSize > (int64_t)nMaxCacheIncreaseSinceLastFlush)
```

Comment:
no actually related to the PR, but why `nCoinCacheMaxSize` is declared as `int64_t` in `global.cpp` ? Is there a case where it can assume a negative value?